### ISSUE TITLE
Codepush release-react doesn't find Android app id.

### ISSUE THREAD

**Issue by xavieramoros** [microsoft/appcenter-cli#684]
Codepush react-react doesn't find the android app id.

Running:
>appcenter codepush release-react -a adrianfernandez/SpotaHome-Android

or with the full command as well:
> appcenter codepush release-react -a "adrianfernandez/SpotaHome-Android" -d Production --target-binary-version "4 - 4.6.3" --mandatory

returns the following error:

>Error: Command 'codepush release-react -a adrianfernandez/SpotaHome-Android' failed with exception "' adrianfernandez/SpotaHome-Android' is not a valid application id"

The application id seems correct, it is shown in the output of `appcenter apps list`:

![image](https://user-images.githubusercontent.com/347030/67210108-cdc2d100-f418-11e9-8420-ca602ed68f85.png)

iOS seems fine, I was able to do use codepush using:
>appcenter codepush release-react -a adrianfernandez/SpotaHome-iOS -d Production --target-binary-version "4 - 4.6.3" --plist-file='ios/native/Info.plist' --mandatory

<img width="1440" alt="Screenshot 2019-10-21 at 15 50 41" src="https://user-images.githubusercontent.com/347030/67211296-c8668600-f41a-11e9-8086-8a5917267be3.png">

I've successfully used codepush in the past with a very similar command:
>appcenter codepush release-react -a adrianfernandez/SpotaHome-Android -d Production --target-binary-version '3.6.0 - 3.7.1' --mandatory

![Screenshot 2019-10-21 at 15 56 39](https://user-images.githubusercontent.com/347030/67211772-93a6fe80-f41b-11e9-8653-eb01c7d9f819.png)

Using appcenter-cli@2.2.1
React Native: 0.59.10
react-native-code-push: "^5.6.0"

**Comment by xavieramoros** [comment]
I also tried to downgrade "appcenter-cli" all the way down to 1.2.2 and the problem still persists.

**Comment by xavieramoros** [comment]
Alright, so I tried to run locally `appcenter-cli` and it actually worked. 

I did:
-cloned repo (github.com/microsoft/appcenter-cli)
-npm install
-npm run build
and then from my react-native project run:
> node ../appcenter-cli/dist/index.js codepush release-react -a adrianfernandez/SpotaHome-Android -d Production --target-binary-version "4 - 4.6.3" --mandatory --debug

Looking at the appcenter dashboard, the codepush release is there correctly now:
![image](https://user-images.githubusercontent.com/347030/67229928-9a923900-f43c-11e9-8db4-c212874f01a3.png)


Not sure what is the commit that makes it work from master and not from the latest 2.2.1 release.

**Comment by jerietve** [comment]
Thank you for the detailed report. I'm glad you found a workaround but will keep the bug open to investigate it further.

**Comment by xavieramoros** [comment]
Follow up on the issue (better late than never ;)):

### TLDR
We can close the bug, I cannot reproduce it anymore.

### Debugging the issue

* This wasn't working for me:
> appcenter codepush release-react -a adrianfernandez/SpotaHome-Android

>Error: Command 'codepush release-react -a adrianfernandez/SpotaHome-Android' failed with exception "' adrianfernandez/SpotaHome-Android' is not a valid application id"

It looks like there is an unwanted space at the beginning of the `appName`.

* This worked
> appcenter codepush release-react -a=adrianfernandez/SpotaHome-Android

The difference is in the command params `-a <ownerName>/<appName>` vs `-a=<ownerName>/<appName>`.

### Appcenter cli code

In this line: 
https://github.com/microsoft/appcenter-cli/blob/master/src/util/commandline/app-command.ts#L24 

`this.appOption` has already an incorrect value (with unwanted spacing), so the problem comes probably from the cli argument parsing that happens here:
https://github.com/microsoft/appcenter-cli/blob/master/src/util/commandline/command.ts#L41

### Conclusion
After finding the location of the issue, I was trying to debug a bit more and now the original command (with `-a <ownerName>/<appName>`) works correctly, I cannot reproduce anymore my original issue.

So I guess we can close this bug (finally).

**Comment by kopax-polyconseil** [comment]
How can you find `ENVIRONMENT` ?  I have tried : 

```
px appcenter codepush deployment list --app $APPCENTER_PROJECT_ID
Error: Failed to get list of deployments for the app
````

**Comment by DmitriyKirakosyan** [comment]
Hi @kopax-polyconseil , you need to create a new deployment first. You can do it on the AppCenter Portal by navigating "your app -> Distribute -> CodePush" and make sure you have "Production" deployment there. Otherwise just create one.
